File: backend/src/main/java/com/example/realestate/model/Eoi.java
package com.example.realestate.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "expressions_of_interest")
public class Eoi {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long investorId;
    private Long projectId;
    private String status;
    private LocalDateTime submissionDate;

    public Eoi() {
        this.submissionDate = LocalDateTime.now();
        this.status = "SUBMITTED";
    }

    public Eoi(Long investorId, Long projectId) {
        this();
        this.investorId = investorId;
        this.projectId = projectId;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getInvestorId() { return investorId; }
    public void setInvestorId(Long investorId) { this.investorId = investorId; }

    public Long getProjectId() { return projectId; }
    public void setProjectId(Long projectId) { this.projectId = projectId; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public LocalDateTime getSubmissionDate() { return submissionDate; }
    public void setSubmissionDate(LocalDateTime submissionDate) { this.submissionDate = submissionDate; }
}

File: lib/models/eoi.dart
class Eoi {
  final int? id;
  final int investorId;
  final int projectId;
  final String status;
  final DateTime? submissionDate;

  Eoi({
    this.id,
    required this.investorId,
    required this.projectId,
    this.status = 'SUBMITTED',
    this.submissionDate,
  });

  factory Eoi.fromJson(Map<String, dynamic> json) {
    return Eoi(
      id: json['id'],
      investorId: json['investorId'],
      projectId: json['projectId'],
      status: json['status'] ?? 'SUBMITTED',
      submissionDate: json['submissionDate'] != null 
          ? DateTime.parse(json['submissionDate']) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'investorId': investorId,
      'projectId': projectId,
      'status': status,
    };
  }
}
